### PR TITLE
[Enhancement](Nereids) Automatic compute logical properties

### PR DESCRIPTION
**Pull request by 924060929**
# Proposed changes

Issue Number: no issue

## Problem Summary:

Automatic compute logical properties

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: Yes
3. Has document been added or modified: No Need
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java [hunk 1]
```diff
@@ -39,18 +41,29 @@
...
         Preconditions.checkArgument(children.size() == 2);
-        return new LogicalBinaryPlan(operator, groupExpression.orElse(null),
-                logicalProperties, children.get(0), children.get(1));
+        return new LogicalBinaryPlan(operator, groupExpression, Optional.empty(),
+            children.get(0), children.get(1));
+    }
+
+    @Override
+    public LogicalBinaryPlan<OP_TYPE, LEFT_CHILD_TYPE, RIGHT_CHILD_TYPE> withOutput(List<Slot> output) {
+        LogicalProperties logicalProperties = new LogicalProperties(output);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java:66]: maybe we need a function like withOutput for LogicalProperties

File: fe/fe-core/src/main/java/org/apache/doris/nereids/operators/plans/logical/LogicalOperator.java [hunk 2]
```diff
@@ -27,5 +27,5 @@
  * interface for all concrete logical plan operator.
  */
 public interface LogicalOperator extends PlanOperator {
-    List<Slot> computeOutput(Plan... inputs);
+    List<Slot> computeOutput(List<Plan> inputs);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/operators/plans/logical/LogicalOperator.java:30]: could it better to replace `computeOutput` with `computeLogicalProperties`?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java:66]: LogicalLeaf(UnboundRelation).getLogicalProperties() will throw UnboundException.
So can not invoke getLogicalProperties().withOutput(newOutput);

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java:66]: how can i do only update output in logical properties and leave the other attributes unchanged?

Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalBinaryPlan.java:66]: maybe new LogicalProperties(newOutput, operator.computeOtherFields()) ?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/operators/plans/logical/LogicalOperator.java (hunk 2 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/operators/plans/logical/LogicalOperator.java]: done

**Review by morrySnow — DISMISSED** [review]
LGTM

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**